Base/src/main/java/io/deephaven/base/clock/Clock.java

```diff
@@ -3,17 +3,21 @@
  */
 package io.deephaven.base.clock;
 
+import io.deephaven.clock.RealTimeClock;
+import org.jetbrains.annotations.NotNull;
+
 public interface Clock {
 
     long currentTimeMillis();
 
     long currentTimeMicros();
 
-    interface Factory {
-        Clock getClock();
-    }
+    long currentTimeNanos();
+
+    final class Null implements Clock {
```

Comment:
enum is a better pattern IMO, but I noticed there aren't any real users of Null.